using-mobile-extensions/adobe-experience-platform-messaging/messaging-api-reference.md

```diff
@@ -0,0 +1,420 @@
+{% tab title="Android" %}
+### Java
+
+#### Syntax
+
+```java
+public AEPMessagingFCMPushPayload(RemoteMessage message)
+```
+
+* *message* - `RemoteMessage` message containing the payload data with the necessary attributes for creating push notification
+
+```java
+public AEPMessagingFCMPushPayload(Map<String, String> data)
+```
+
+* *data* - `Map<String, String>` data payload containing the necessary attributes for creating push notification
+
+#### Examples
+
+```java
+// Using the remote message
+@Override
+public void onMessageReceived(@NonNull RemoteMessage remoteMessage) {
+    AEPMessagingFCMPushPayload payload = new AEPMessagingFCMPushPayload(remoteMessage);
+}
+
+// Using the data map
+@Override
+public void onMessageReceived(@NonNull RemoteMessage remoteMessage) {
+    AEPMessagingFCMPushPayload payload = new AEPMessagingFCMPushPayload(remoteMessage.getData());
+}
+```
+
+{% endtab %}
+{% endtabs %}
+
+### Public APIs
+Public APIs for getting attributes from push payload which are used while creating the push notification.
+
+{% tabs %}
+{% tab title="Android" %}
+
+### Java
+
+#### Syntax
+
+```java
+// Returns the title from the remote message
+public String getTitle()
+
+// Returns the body from the remote message 
+public String getBody()
+
+// Returns the sound from the remote message 
+// The sound string represents the filename of a sound resource bundled in the app.
+public String getSound()
+
+// Returns the notification badge count from the remote message 
+public int getBadgeCount()
+
+// Returns the notification priority from the remote message. 
+// Check out the firebase documentation here (https://firebase.google.com/docs/reference/fcm/rest/v1/projects.messages#notificationpriority) 
+public int getNotificationPriority()
+
+// Returns the channel Id from the remote message. 
+public String getChannelId()
+
+// Returns the icon string from the remote message.
+// The icon string represents the drawable resource name in the app.
+public String getIcon()
+
+// Returns the image url from the remote message.
+// The icon string represents the drawable resource name in the app.
+public String getImageUrl()
+
+// Returns the data map from the remote message.
+public Map<String, String> getData()
+
+// Returns an ActionType object which represents the type of action which needs to be performed on push notification interaction.
+// Check the ActionType enum definition below.
+public ActionType getActionType()
+
+// Returns a String action uri which is used to direct the push notification interaction.
+public String getActionUri()
+
+// Returns a list of ActionButtons. Check the ActionButtons class definition below
+public List<ActionButton> getActionButtons()
+```
+
+##### Internal Classes and Enums
+
+###### ActionType
+
+```java
+public enum ActionType {
+    DEEPLINK, WEBURL, DISMISS, NONE
+}
+```
+
+###### ActionButtons
+```java
+// Constructor
+public ActionButton(final String label, final String link, final String type)
```

Comment:
is the third param supposed to be `String`, or `ActionType`?

Reply:
Its the String we get from payload and then we convert it to ActionType